File: scripts/fetch_logos.mjs
// scripts/fetch_logos.mjs — Download all pro-league team logos from ESPN's
// CDN into ../public/teams/{league}/{slug}.png. Run from the project root:
//     node scripts/fetch_logos.mjs
// or via the npm alias:
//     npm run logos
//
// No dependencies — uses Node's built-in fetch + fs. Skips files that
// already exist on disk so re-runs are idempotent / fast.
//
// If you want to re-download everything, delete public/teams/ first.

import { writeFile, mkdir, stat } from "node:fs/promises";
import { join, dirname } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const ROOT = join(__dirname, "..", "public", "teams");

// Each entry: [fullName, slug, abbr]. Aliases (multiple names → same slug)
// are filtered to a single download per slug. Mirror of src/teams.js mapping.
const TEAMS = {
  nba: [
    ["Atlanta Hawks", "atl"], ["Boston Celtics", "bos"], ["Brooklyn Nets", "bkn"],
    ["Charlotte Hornets", "cha"], ["Chicago Bulls", "chi"], ["Cleveland Cavaliers", "cle"],
    ["Dallas Mavericks", "dal"], ["Denver Nuggets", "den"], ["Detroit Pistons", "det"],
    ["Golden State Warriors", "gs"], ["Houston Rockets", "hou"], ["Indiana Pacers", "ind"],
    ["LA Clippers", "lac"], ["Los Angeles Lakers", "lal"], ["Memphis Grizzlies", "mem"],
    ["Miami Heat", "mia"], ["Milwaukee Bucks", "mil"], ["Minnesota Timberwolves", "min"],
    ["New Orleans Pelicans", "no"], ["New York Knicks", "ny"], ["Oklahoma City Thunder", "okc"],
    ["Orlando Magic", "orl"], ["Philadelphia 76ers", "phi"], ["Phoenix Suns", "phx"],
    ["Portland Trail Blazers", "por"], ["Sacramento Kings", "sac"], ["San Antonio Spurs", "sa"],
    ["Toronto Raptors", "tor"], ["Utah Jazz", "utah"], ["Washington Wizards", "wsh"],
  ],
  nfl: [
    ["Arizona Cardinals", "ari"], ["Atlanta Falcons", "atl"], ["Baltimore Ravens", "bal"],
    ["Buffalo Bills", "buf"], ["Carolina Panthers", "car"], ["Chicago Bears", "chi"],
    ["Cincinnati Bengals", "cin"], ["Cleveland Browns", "cle"], ["Dallas Cowboys", "dal"],
    ["Denver Broncos", "den"], ["Detroit Lions", "det"], ["Green Bay Packers", "gb"],
    ["Houston Texans", "hou"], ["Indianapolis Colts", "ind"], ["Jacksonville Jaguars", "jax"],
    ["Kansas City Chiefs", "kc"], ["Las Vegas Raiders", "lv"], ["Los Angeles Chargers", "lac"],
    ["Los Angeles Rams", "lar"], ["Miami Dolphins", "mia"], ["Minnesota Vikings", "min"],
    ["New England Patriots", "ne"], ["New Orleans Saints", "no"], ["New York Giants", "nyg"],
    ["New York Jets", "nyj"], ["Philadelphia Eagles", "phi"], ["Pittsburgh Steelers", "pit"],
    ["San Francisco 49ers", "sf"], ["Seattle Seahawks", "sea"], ["Tampa Bay Buccaneers", "tb"],
    ["Tennessee Titans", "ten"], ["Washington Commanders", "wsh"],
  ],
  mlb: [
    ["Arizona Diamondbacks", "ari"], ["Atlanta Braves", "atl"], ["Baltimore Orioles", "bal"],
    ["Boston Red Sox", "bos"], ["Chicago Cubs", "chc"], ["Chicago White Sox", "chw"],
    ["Cincinnati Reds", "cin"], ["Cleveland Guardians", "cle"], ["Colorado Rockies", "col"],
    ["Detroit Tigers", "det"], ["Houston Astros", "hou"], ["Kansas City Royals", "kc"],
    ["Los Angeles Angels", "laa"], ["Los Angeles Dodgers", "lad"], ["Miami Marlins", "mia"],
    ["Milwaukee Brewers", "mil"], ["Minnesota Twins", "min"], ["New York Mets", "nym"],
    ["New York Yankees", "nyy"], ["Oakland Athletics", "oak"], ["Philadelphia Phillies", "phi"],
    ["Pittsburgh Pirates", "pit"], ["San Diego Padres", "sd"], ["Seattle Mariners", "sea"],
    ["San Francisco Giants", "sf"], ["St. Louis Cardinals", "stl"], ["Tampa Bay Rays", "tb"],
    ["Texas Rangers", "tex"], ["Toronto Blue Jays", "tor"], ["Washington Nationals", "wsh"],
  ],
  nhl: [
    ["Anaheim Ducks", "ana"], ["Boston Bruins", "bos"], ["Buffalo Sabres", "buf"],
    ["Calgary Flames", "cgy"], ["Carolina Hurricanes", "car"], ["Chicago Blackhawks", "chi"],
    ["Colorado Avalanche", "col"], ["Columbus Blue Jackets", "cbj"], ["Dallas Stars", "dal"],
    ["Detroit Red Wings", "det"], ["Edmonton Oilers", "edm"], ["Florida Panthers", "fla"],
    ["Los Angeles Kings", "la"], ["Minnesota Wild", "min"], ["Montreal Canadiens", "mtl"],
    ["Nashville Predators", "nsh"], ["New Jersey Devils", "nj"], ["New York Islanders", "nyi"],
    ["New York Rangers", "nyr"], ["Ottawa Senators", "ott"], ["Philadelphia Flyers", "phi"],
    ["Pittsburgh Penguins", "pit"], ["San Jose Sharks", "sj"], ["Seattle Kraken", "sea"],
    ["St. Louis Blues", "stl"], ["Tampa Bay Lightning", "tb"], ["Toronto Maple Leafs", "tor"],
    ["Utah Hockey Club", "utah"], ["Vancouver Canucks", "van"], ["Vegas Golden Knights", "vgs"],
    ["Washington Capitals", "wsh"], ["Winnipeg Jets", "wpg"],
  ],
};

async function exists(p) {
  try { const s = await stat(p); return s.size > 200; } catch { return false; }
}

async function downloadOne(league, slug, name) {
  const dest = join(ROOT, league, `${slug}.png`);
  if (await exists(dest)) return { status: "skip", name };
  const url = `https://a.espncdn.com/i/teamlogos/${league}/500/${slug}.png`;
  try {
    const r = await fetch(url, { headers: { "User-Agent": "Mozilla/5.0" } });
    if (!r.ok) return { status: `http-${r.status}`, name };
    const buf = Buffer.from(await r.arrayBuffer());
    if (buf.length < 200) return { status: `too-small-${buf.length}B`, name };
    await writeFile(dest, buf);
    return { status: `ok-${buf.length}B`, name };
  } catch (e) {
    return { status: `err: ${e.message}`, name };
  }
}

async function main() {
  let ok = 0, skip = 0;
  const fails = [];
  for (const [league, rows] of Object.entries(TEAMS)) {
    await mkdir(join(ROOT, league), { recursive: true });
    const seen = new Set();
    // Run downloads for this league in parallel (max ~30 in flight)
    const tasks = rows
      .filter(([, slug]) => { if (seen.has(slug)) return false; seen.add(slug); return true; })
      .map(([name, slug]) => downloadOne(league, slug, name).then(r => ({ league, slug, ...r })));
    const results = await Promise.all(tasks);
    for (const r of results) {
      const tag = r.status.startsWith("ok") ? "OK" : r.status === "skip" ? "--" : "FAIL";
      console.log(`  [${tag}] ${r.league}/${r.slug}.png  ${r.name}  ${r.status}`);
      if (r.status.startsWith("ok")) ok++;
      else if (r.status === "skip") skip++;
      else fails.push(r);
    }
  }
  console.log(`\n=== ${ok} downloaded, ${skip} already present, ${fails.length} failed ===`);
  if (fails.length) {
    console.log("\nFailures (the page will still work — these teams render the monogram fallback):");
    for (const f of fails) console.log(`  ${f.league}/${f.slug}.png  ${f.name}  ->  ${f.status}`);
    process.exit(2);
  }
}

main().catch(e => { console.error(e); process.exit(1); });

File: scripts/fetch_logos.py
#!/usr/bin/env python3
"""Download all pro-league team logos from ESPN CDN into BetSimV2/public/teams/{league}/{slug}.png."""
import os, sys, json, urllib.request, urllib.error

import pathlib
ROOT = str(pathlib.Path(__file__).resolve().parent.parent / "public" / "teams")

# Each entry: full_name, slug (ESPN's URL slug), abbr (display)
TEAMS = {
    "nba": [
        ("Atlanta Hawks", "atl", "ATL"),
        ("Boston Celtics", "bos", "BOS"),
        ("Brooklyn Nets", "bkn", "BKN"),
        ("Charlotte Hornets", "cha", "CHA"),
        ("Chicago Bulls", "chi", "CHI"),
        ("Cleveland Cavaliers", "cle", "CLE"),
        ("Dallas Mavericks", "dal", "DAL"),
        ("Denver Nuggets", "den", "DEN"),
        ("Detroit Pistons", "det", "DET"),
        ("Golden State Warriors", "gs", "GSW"),
        ("Houston Rockets", "hou", "HOU"),
        ("Indiana Pacers", "ind", "IND"),
        ("LA Clippers", "lac", "LAC"),
        ("Los Angeles Clippers", "lac", "LAC"),  # alt name
        ("Los Angeles Lakers", "lal", "LAL"),
        ("Memphis Grizzlies", "mem", "MEM"),
        ("Miami Heat", "mia", "MIA"),
        ("Milwaukee Bucks", "mil", "MIL"),
        ("Minnesota Timberwolves", "min", "MIN"),
        ("New Orleans Pelicans", "no", "NOP"),
        ("New York Knicks", "ny", "NYK"),
        ("Oklahoma City Thunder", "okc", "OKC"),
        ("Orlando Magic", "orl", "ORL"),
        ("Philadelphia 76ers", "phi", "PHI"),
        ("Phoenix Suns", "phx", "PHX"),
        ("Portland Trail Blazers", "por", "POR"),
        ("Sacramento Kings", "sac", "SAC"),
        ("San Antonio Spurs", "sa", "SAS"),
        ("Toronto Raptors", "tor", "TOR"),
        ("Utah Jazz", "utah", "UTA"),
        ("Washington Wizards", "wsh", "WAS"),
    ],
    "nfl": [
        ("Arizona Cardinals", "ari", "ARI"),
        ("Atlanta Falcons", "atl", "ATL"),
        ("Baltimore Ravens", "bal", "BAL"),
        ("Buffalo Bills", "buf", "BUF"),
        ("Carolina Panthers", "car", "CAR"),
        ("Chicago Bears", "chi", "CHI"),
        ("Cincinnati Bengals", "cin", "CIN"),
        ("Cleveland Browns", "cle", "CLE"),
        ("Dallas Cowboys", "dal", "DAL"),
        ("Denver Broncos", "den", "DEN"),
        ("Detroit Lions", "det", "DET"),
        ("Green Bay Packers", "gb", "GB"),
        ("Houston Texans", "hou", "HOU"),
        ("Indianapolis Colts", "ind", "IND"),
        ("Jacksonville Jaguars", "jax", "JAX"),
        ("Kansas City Chiefs", "kc", "KC"),
        ("Las Vegas Raiders", "lv", "LV"),
        ("Los Angeles Chargers", "lac", "LAC"),
        ("Los Angeles Rams", "lar", "LAR"),
        ("Miami Dolphins", "mia", "MIA"),
        ("Minnesota Vikings", "min", "MIN"),
        ("New England Patriots", "ne", "NE"),
        ("New Orleans Saints", "no", "NO"),
        ("New York Giants", "nyg", "NYG"),
        ("New York Jets", "nyj", "NYJ"),
        ("Philadelphia Eagles", "phi", "PHI"),
        ("Pittsburgh Steelers", "pit", "PIT"),
        ("San Francisco 49ers", "sf", "SF"),
        ("Seattle Seahawks", "sea", "SEA"),
        ("Tampa Bay Buccaneers", "tb", "TB"),
        ("Tennessee Titans", "ten", "TEN"),
        ("Washington Commanders", "wsh", "WAS"),
    ],
    "mlb": [
        ("Arizona Diamondbacks", "ari", "ARI"),
        ("Atlanta Braves", "atl", "ATL"),
        ("Baltimore Orioles", "bal", "BAL"),
        ("Boston Red Sox", "bos", "BOS"),
        ("Chicago Cubs", "chc", "CHC"),
        ("Chicago White Sox", "chw", "CHW"),
        ("Cincinnati Reds", "cin", "CIN"),
        ("Cleveland Guardians", "cle", "CLE"),
        ("Colorado Rockies", "col", "COL"),
        ("Detroit Tigers", "det", "DET"),
        ("Houston Astros", "hou", "HOU"),
        ("Kansas City Royals", "kc", "KC"),
        ("Los Angeles Angels", "laa", "LAA"),
        ("Los Angeles Dodgers", "lad", "LAD"),
        ("Miami Marlins", "mia", "MIA"),
        ("Milwaukee Brewers", "mil", "MIL"),
        ("Minnesota Twins", "min", "MIN"),
        ("New York Mets", "nym", "NYM"),
        ("New York Yankees", "nyy", "NYY"),
        ("Oakland Athletics", "oak", "OAK"),
        ("Athletics", "oak", "ATH"),  # post-rebrand
        ("Philadelphia Phillies", "phi", "PHI"),
        ("Pittsburgh Pirates", "pit", "PIT"),
        ("San Diego Padres", "sd", "SD"),
        ("Seattle Mariners", "sea", "SEA"),
        ("San Francisco Giants", "sf", "SF"),
        ("St. Louis Cardinals", "stl", "STL"),
        ("St Louis Cardinals", "stl", "STL"),  # alt punctuation
        ("Tampa Bay Rays", "tb", "TB"),
        ("Texas Rangers", "tex", "TEX"),
        ("Toronto Blue Jays", "tor", "TOR"),
        ("Washington Nationals", "wsh", "WSH"),
    ],
    "nhl": [
        ("Anaheim Ducks", "ana", "ANA"),
        ("Boston Bruins", "bos", "BOS"),
        ("Buffalo Sabres", "buf", "BUF"),
        ("Calgary Flames", "cgy", "CGY"),
        ("Carolina Hurricanes", "car", "CAR"),
        ("Chicago Blackhawks", "chi", "CHI"),
        ("Colorado Avalanche", "col", "COL"),
        ("Columbus Blue Jackets", "cbj", "CBJ"),
        ("Dallas Stars", "dal", "DAL"),
        ("Detroit Red Wings", "det", "DET"),
        ("Edmonton Oilers", "edm", "EDM"),
        ("Florida Panthers", "fla", "FLA"),
        ("Los Angeles Kings", "la", "LAK"),
        ("Minnesota Wild", "min", "MIN"),
        ("Montreal Canadiens", "mtl", "MTL"),
        ("Nashville Predators", "nsh", "NSH"),
        ("New Jersey Devils", "nj", "NJ"),
        ("New York Islanders", "nyi", "NYI"),
        ("New York Rangers", "nyr", "NYR"),
        ("Ottawa Senators", "ott", "OTT"),
        ("Philadelphia Flyers", "phi", "PHI"),
        ("Pittsburgh Penguins", "pit", "PIT"),
        ("San Jose Sharks", "sj", "SJ"),
        ("Seattle Kraken", "sea", "SEA"),
        ("St. Louis Blues", "stl", "STL"),
        ("St Louis Blues", "stl", "STL"),
        ("Tampa Bay Lightning", "tb", "TB"),
        ("Toronto Maple Leafs", "tor", "TOR"),
        ("Utah Hockey Club", "utah", "UTA"),
        ("Utah Mammoth", "utah", "UTA"),  # rumored 2026 rebrand fallback
        ("Vancouver Canucks", "van", "VAN"),
        ("Vegas Golden Knights", "vgs", "VGK"),
        ("Washington Capitals", "wsh", "WSH"),
        ("Winnipeg Jets", "wpg", "WPG"),
    ],
}


def download(url, dest):
    if os.path.exists(dest) and os.path.getsize(dest) > 0:
        return "skip"
    try:
        req = urllib.request.Request(url, headers={"User-Agent": "Mozilla/5.0"})
        with urllib.request.urlopen(req, timeout=10) as r:
            data = r.read()
        if len(data) < 200:
            return f"too-small ({len(data)}B)"
        with open(dest, "wb") as f:
            f.write(data)
        return f"ok ({len(data)}B)"
    except urllib.error.HTTPError as e:
        return f"http-{e.code}"
    except Exception as e:
        return f"err: {e}"


def main():
    results = {"ok": 0, "skip": 0, "fail": []}
    seen = set()
    for league, teams in TEAMS.items():
        league_dir = os.path.join(ROOT, league)
        os.makedirs(league_dir, exist_ok=True)
        for name, slug, abbr in teams:
            key = (league, slug)
            if key in seen:
                continue  # skip duplicate slugs from alt names
            seen.add(key)
            url = f"https://a.espncdn.com/i/teamlogos/{league}/500/{slug}.png"
            dest = os.path.join(league_dir, f"{slug}.png")
            status = download(url, dest)
            tag = "OK" if status.startswith(("ok", "skip")) else "FAIL"
            print(f"  [{tag}] {league}/{slug}.png ({name}) -> {status}")
            if status.startswith("ok"):
                results["ok"] += 1
            elif status == "skip":
                results["skip"] += 1
            else:
                results["fail"].append((league, slug, name, status))
    print()
    print(f"=== summary: {results['ok']} downloaded, {results['skip']} skipped, {len(results['fail'])} failed ===")
    for f in results["fail"]:
        print(" FAIL:", f)
    return 0 if not results["fail"] else 2


if __name__ == "__main__":
    sys.exit(main())
